File: examples/programming/move_to.js
"use strict"

var botnana = require('../../index')

function test() {
    // Show reponse data
    botnana.debug_level = 1;
    botnana.once("ready", function() {
        // Programming API                
        var Program = botnana.Program;
        // Program p
        var p = new Program("p");
        var s1 = p.ethercat.slave(1);
        s1.move_to(50000);
        p.deploy();
        // Run Program p1
        botnana.once("deployed", function() { p.run(); });
    })
    botnana.start("ws://192.168.1.117:3012");
}

test();

File: botnanars/examples/move_to.rs
extern crate botnanars;
use botnanars::botnana;
use botnanars::programmed::Program;
use std::{thread, time};
use std::sync::{Arc, Mutex};

fn main() {
    let mut botnana = Arc::new(Mutex::new(botnana::botnana::new().unwrap()));

    let btn = botnana.clone();

    botnana.lock().unwrap().once("ready", move |msg| {
        let mut p = btn.lock().unwrap().program("p");
        let s1 = p.ethercat.slave(1);
        match s1 {
            Some(mut s) => {
                s.move_to(50000);
            }
            None => {}
        }

        p.deploy();         
    });

    let btn = botnana.clone();
    botnana.lock().unwrap().once("deployed", move |msg| {
        let p = btn.lock().unwrap().program("p");
        p.run();
    });

    botnana.lock().unwrap().start("ws://localhost:3012");
    loop {}
}